frontend/kesaseteli/employer/src/components/header/Header.tsx

```diff
@@ -59,12 +52,12 @@ const Header: React.FC = () => {
       login={
         !isLoading
           ? {
-              isAuthenticated: !isLoginPage && userQuery.isSuccess,
+              isAuthenticated: !isLoginPage && isSuccess,
               loginLabel: t('common:header.loginLabel'),
               logoutLabel: t('common:header.logoutLabel'),
               onLogin: login,
               onLogout: logout,
-              userName: userQuery.isSuccess ? userQuery.data.name : undefined,
+              userName: isSuccess ? data?.name : undefined,
```

Comment:
here you can see that type narrowing didnt work and you have to use question mark `data?` even when data should be not empty.